app/src/main/kotlin/compiler/ast/Statement.kt

```diff
@@ -52,6 +52,13 @@ sealed class Statement : AstNode {
         override val location: LocationRange? = null,
     ) : Statement()
 
+    data class ForeachLoop(
+        val receivingVariable: Variable,
```

Comment:
This should be `String`, but I guess in this way the implementation will be somewhat easier.